README.md

```diff
@@ -1,223 +1,350 @@
+      - [2.4.1 `interop-cpp`](#241--interop-cpp-)
+      - [2.4.2 `interop-cpp-no-cargo`](#242--interop-cpp-no-cargo-)
+      - [2.4.3 `ping-pong-with-noise`](#243--ping-pong-with-noise-)
+      - [2.4.4 `ping-pong-without-noise`](#244--ping-pong-without-noise-)
+      - [2.4.5 `sv1-client-and-server`](#245--sv1-client-and-server-)
+      - [2.4.6 `sv2-proxy`](#246--sv2-proxy-)
+      - [2.4.7 `template-provider-test`](#247--template-provider-test-)
+    + [2.5 `test`](#25--test-)
+    + [2.6 `experimental`](#26--experimental-)
+  * [3. Branches](#3-branches)
+
+
+## 1. Goals
 
 The goal of the project is to provide:
-* A robust set of Sv2 primitives as rust library crates that can be used by anyone that want expand
-    the protocol or implement a role (EG a pool that want to support Sv2, or mining-device producer
-    that want to integrate Sv2 into the firmware, a bitcoin node that want to implement Template Provide capabilities).
-    Make this primitives available also to non rust user. 
-* A set of helpers built on top of the above primitives and external bitcoin related rust crates,
-    that can be used by anyone the want to implement an Sv2 role (EG a pool that want to support Sv2
-    and that use rust)
-* An open-source implementation of an Sv2 proxy, that can be used by miners that want to use Sv2.
-* An open-source implementation of an Sv2 pool that can be used by anyone that want to operate an Sv2
-    pool.
-* A self hosted Job Negotiator + Template Provider that can be used by Sv2 or Sv1 pool that want to
-    offer job negotiation capabilities to their costumer without messing with the pool codebase (a
-    minimum amount of modification is still required)
-
-## STRUCTURE
-
-This workspace is divided in 6 macro-areas, (1) *protocols* several Stratum library, (2) *roles* the
-actual Sv2 roles, (3) *utils*, (4) *examples*, (5) *test* integration tests, (6) *experimental* not yet
-specified part of the protocol or extensions.
-
-Every dependency related to benchmarking and testing can be always opted out so we never list
-them under **External dependencies**.
-
-### PROTOCOLS
-
-Under *protocols* there is *v1* that contain an Sv1 library and *v2* that contain several Sv2
-libraries.
-
-### PROTOCOLS/V1
-TODO
-
-### PROTOCOLS/V2
-TODO
-
-### PROTOCOLS/V2/CONST-SV2
-
-A bunch of Sv2 related constants.
+* These libraries can all be used, or only a subset of them can be used, depending on the
+  functionality desired by the end user. Some examples of individuals who benefit from the
+  libraries in this repository are as follows (note that not all features are completed yet):
+    * A miner who runs a mining farm with Sv2-compatible mining firmware mining to a 
+      Sv2-compatible pool can use this library as a proxy which allows them to use the Group
+      Channels, which reduce their bandwidth consumption and bolster their efficiency. Standard
+      Channels are also supported in here, but are not as efficient as Group Channels.
```

Comment:
I wouldn't talk about Standard Channel vs Group Channel (cause they are complementary).
To talk about Standard Channel vs Group Channel we need to introduce:
1. HOM and non HOM nodes
2. mining device
3. pool
Then we can say that the proxy do support to mine with a HOM pool but is less efficient that mine with a non HOM pool.